tex2pdf_service/tests/test_docker.py

```diff
@@ -6,12 +6,13 @@
 import pytest
 from bin.compile_submissions import get_outcome_meta
 import logging
+import json
 
 PORT = 33031
 
-def submit_tarball(service: str, tarball: str, outcome_file: str, tex2pdf_timeout: int = 30, post_timeout: int = 10) -> None | dict:
+def submit_tarball(service: str, tarball: str, outcome_file: str, tex2pdf_timeout: int = 30, post_timeout: int = 10, json_response: bool = False, api_args: str = "") -> None | dict:
     meta = None
-    url = service + f"/?timeout={tex2pdf_timeout}"
+    url = service + f"/?timeout={tex2pdf_timeout}{api_args}"
```

Comment:
Not sure. I am fine with both

Reply:
For reference, there's a module: [urllib.parse.urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)

Reply:
Not sure that is needed here, since we control what makes up the api call, so we know that we don't have spaces to quote etc. IMHO that would only obfuscate things.

Concerning the args, do you think that is better:
```
    if not api_args:
        # check for empty string or empty list
        final_api_args = ""
    else:
        if isinstance(api_args, str):
            if api_args.startswith('&'):
                final_api_args = api_args
            else:
                final_api_args = '&' + api_args
        else:
            final_api_args = "&".join(["", *api_args])
```
I am not sure ... it is not for public consumption but testing.

Reply:
Well, maybe go with a dict as argument and do
```
    params_dict = { "timeout": tex2pdf_timeout }
    params_dict.update(api_args)
    params = urllib.parse.urlencode(params_dict)
    url = f"{service}/?{params}"
```
Will do that.

Reply:
Done that now.